arnglld/src/main.rs

```diff
@@ -33,19 +40,175 @@ struct Opt {
     #[clap(short, long, parse(from_occurrences))]
     verbose: usize,
 
+    /// List audio devices
+    #[clap(long)]
+    list_devices: bool,
+
     #[clap(short, long)]
-    callsign: HamAddr,
+    callsign: Option<HamAddr>,
+
+    #[clap(long)]
+    input_audio_device: Option<String>,
+
+    #[clap(long)]
+    output_audio_device: Option<String>,
+}
+
+impl Opt {
+    fn get_output_device(&self) -> Result<cpal::Device, anyhow::Error> {
+        let host = cpal::default_host();
+
+        if let Some(name) = self.output_audio_device.as_ref() {
+            let lc_name = name.to_lowercase();
+
+            // First try exact matches
+            for device in host.output_devices()? {
+                let dev_name = device.name()?.to_lowercase();
```

Comment:
Is early-return really what we want if `name` is not happy? Perhaps we simply skip over that entry?

Reply:
Yeah, not ideal. Will fix.